### PR TITLE
8315410: Undocumented exceptions in java.text.StringCharacterIterator

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR and [CSR](https://bugs.openjdk.org/browse/JDK-8315558) which is a conformance change to document some exceptions in the specification of java.text.StringCharacterIterator.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8315558](https://bugs.openjdk.org/browse/JDK-8315558) to be approved

### Issues
 * [JDK-8315410](https://bugs.openjdk.org/browse/JDK-8315410): Undocumented exceptions in java.text.StringCharacterIterator (**Bug** - P4)
 * [JDK-8315558](https://bugs.openjdk.org/browse/JDK-8315558): Undocumented exceptions in java.text.StringCharacterIterator (**CSR**)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [58c3f396](https://git.openjdk.org/jdk/pull/15547/files/58c3f39648e8066c4055ef1b2ef935dec1518d37)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15547/head:pull/15547` \
`$ git checkout pull/15547`

Update a local copy of the PR: \
`$ git checkout pull/15547` \
`$ git pull https://git.openjdk.org/jdk.git pull/15547/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15547`

View PR using the GUI difftool: \
`$ git pr show -t 15547`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15547.diff">https://git.openjdk.org/jdk/pull/15547.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15547#issuecomment-1703468399)

### CODE REVIEW

**Comment by AlanBateman** [comment]
@justin-curtis-lu You are correct that it needs a CSR as it is specifying exceptions that weren't previously specified. For the PR you'll need to link it to JDK-8315410, then use "/csr" to add the "csr" label.

File: src/java.base/share/classes/java/text/StringCharacterIterator.java [hunk 1]
```diff
@@ -148,6 +154,8 @@ public char last()
 
     /**
      * Implements CharacterIterator.setIndex() for String.
+     *
+     * @throws IllegalArgumentException if {@code p} is an invalid index
```
Comment (naotoj) [inline src/java.base/share/classes/java/text/StringCharacterIterator.java:158]: It'd be clearer if we clarify what the `invalid` means

**Review by naotoj — APPROVED** [review]
LGTM. Reviewed the CSR too.

**Pull request closed without merging**